### DISCUSSION TITLE
Add Support for Encoding AVIF Images?

### DISCUSSION THREAD

**Discussion by Nicholas-Westby** [SixLabors/ImageSharp#1346] [Ideas] [answered]
I'm not using ImageSharp in production yet (I imagine I will in a year or two), but I thought I'd mention that AVIF seems to be the up-and-comer image file format: https://jakearchibald.com/2020/avif-has-landed/

I think it'd be nifty if ImageSharp supported this format. While it's only really [supported in Chrome](https://caniuse.com/avif) at the moment (and in Firefox when a feature flag is set), the size reduction is drastic when compared to alternatives (e.g., the article shows that an image compressed with JPEG becomes 74KB, with WEB becomes 43KB, and with AVIF becomes 18KB).

It also has the advantage that it seems to be the only format that supports a lossy alpha channel (unless you count http://quasimondo.com/ZorroSVG/, which uses JPEG to compress the alpha channel within an SVG).

I suspect this format will become the main format of choice in a year or so, so better to get it supported sooner rather than later.

**Comment by JimBobSquarePants** [comment, marked as answer]
Yep see #1320 

Huge job but someone’s gotta do it!
